src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/ScanBlobScanLogHybridPollingStrategy.cs

```diff
@@ -0,0 +1,243 @@
+        }
+
+        public void Notify(IStorageBlob blobWritten)
+        {
+            ThrowIfDisposed();
+            _blobsFoundFromScanOrNotification.Enqueue(blobWritten);
+            _pollLogStrategy.Notify(blobWritten);
+        }
+
+        public void Dispose()
+        {
+            if (!_disposed)
+            {
+                _pollLogStrategy.Dispose();
+                _cancellationTokenSource.Dispose();
+                _disposed = true;
+            }
+        }
+
+        private void ThrowIfDisposed()
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(null);
+            }
+        }
+
+        public async Task<IEnumerable<IStorageBlob>> PollNewBlobsAsync(
+            IStorageBlobContainer container, ContainerScanInfo containerScanInfo, CancellationToken cancellationToken)
+        {
+            IEnumerable<IStorageListBlobItem> currentBlobs;
+            IStorageBlobResultSegment blobSegment;
+            int blobPollLimitPerContainer = _scanBlobLimitPerPoll / _scanInfo.Count;
+            BlobContinuationToken continuationToken = containerScanInfo.ContinuationToken;
+
+            // if starting the cycle, keep the current time stamp to be used as curser
+            if (continuationToken == null)
+            {
+                containerScanInfo.CurrentSweepCycleStartTime = DateTime.UtcNow;
+            }
+            try
+            {
+                blobSegment = await container.ListBlobsSegmentedAsync(prefix: null, useFlatBlobListing: true, 
+                    blobListingDetails: BlobListingDetails.None, maxResults: blobPollLimitPerContainer, currentToken: continuationToken, 
+                    options: null, operationContext: null, cancellationToken: cancellationToken);
+                currentBlobs = blobSegment.Results;
+            }
+            catch (StorageException exception)
+            {
+                if (exception.IsNotFound())
+                {
+                    return Enumerable.Empty<IStorageBlob>();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            List<IStorageBlob> newBlobs = new List<IStorageBlob>();
+
+            // Type cast to IStorageBlob is safe due to useFlatBlobListing: true above.
+            foreach (IStorageBlob currentBlob in currentBlobs)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                IStorageBlobProperties properties = currentBlob.Properties;
+                DateTime lastModifiedTimestamp = properties.LastModified.Value.UtcDateTime;
+
+                if (lastModifiedTimestamp > containerScanInfo.LastSweepCycleStartTime)
+                {
+                    newBlobs.Add(currentBlob);
+                }
+            }
+
+            // record continuation token for next chunk retrieval
+            containerScanInfo.ContinuationToken = blobSegment.ContinuationToken;
```

Comment:
I guess in this case here we have to tell customers that there is a budget constraining the number of blobs we poll, and in case of always on keeping that at a minimum enhances performance, and in case of free sites, keeping it at a minimum will ensure functionality